### ISSUE TITLE
Android 12 installation failed

### ISSUE THREAD

**Issue by kubiyuan** [airsdk/Adobe-Runtime-Support#2040]
Apk packaged with airsdk 33.1.1.926 cannot be installed on android12 device

**Comment by ajwfrost** [comment]
Hi

You mention it's an APK file, but then for it to be so large it sounds like an AAB file (which contains 4x runtime libraries, one for each binary architecture, hence the size..)

Are you able to try installing via ADB over USB or similar, to see if there is an error message?

We've got apps building and installing, from both APK and AAB formats, with Android 12 so I'm not sure where the issue is here..

thanks

**Comment by kubiyuan** [comment]
Hi
This is the logcat log while installing

2022-08-01 16:59:53.404 32381-32381/? I/PackageInstaller: PackageInstallObserver error_msg:INSTALL_PARSE_FAILED_MANIFEST_MALFORMED: Failed parse during installPackageLI: /data/app/vmdl201600244.tmp/base.apk (at Binary XML file line #245): com.tencent.mm.sdk.plugin.MMPluginOAuth$Receiver: Targeting S+ (version 31 and above) requires that an explicit value for android:exported be defined when intent filters are present
2022-08-01 16:59:53.405 32381-32381/? I/PackageInstaller: install error code:4

![企业微信截图_1659344551947](https://user-images.githubusercontent.com/54053483/182134218-ebad91aa-5376-4e3b-a215-a09e99412dce.png)

**Comment by kubiyuan** [comment]
When I use this manifest file for packaging

Prompt Me：
/Users/kubi/svn-workspace/MobileFarm/dailybuild/app_config/android-app.xml: error 404: Android attribute android:exported is not allowed to be overridden

<img width="573" alt="image" src="https://user-images.githubusercontent.com/54053483/182137941-68495181-4679-4374-aca7-7818e7ebdeda.png">


Because the xml file could not be uploaded, it was changed to .txt
[android-app.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/9233446/android-app.txt)

**Comment by ajwfrost** [comment]
Hmm, looks like ADT is trying to prevent you from adding the custom intent-handler for the main activity, and as a result it's also not allowing the "exported" attribute to be overridden within the `<activity>` tag.

I'm not sure why there's that restriction, but particularly since there's this change in the requirements, it may be best if we update this...

FYI if you need a quick workaround, you can target the `android-studio` package type with ADT, and then use Android Studio to open the output folder and edit the manifest to add in the necessary `exported` tags manually, and then use their build tools to export a signed bundle..

**Comment by marchbold** [comment]
Just remove the exported flag from your activity with action `android.intent.action.MAIN` and category `LAUNCHER`. 

```xml
<activity android:excludeFromRecents="false" >
    <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
    </intent-filter>
    <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <data android:scheme="qqfarmscheme" android:host="qqfarmioshost"></data>/>
    </intent-filter>
</activity>
```

This activity is special and the exported flag is denied here to make sure this activity is always exported (without that your app wouldn't start normally). AIR has traditionally denied changing this value as this activity is treated specially and merged with the main `AIRAppEntry` activity allowing you to add intent filters for deep links etc to the main activity.

Additionally you have some other odd entries on your other activities, eg in `com.tencent.midas.qq.APMidasQQWalletActivity`:

```
android:exported="false|true"
```

It's likely that wherever you got that from has docs around choosing one value or the other?

**Comment by marchbold** [comment]
@ajwfrost I don't think you need to change this functionality, it's a good safe guard to ensure people don't accidentally not export the main activity.

**Comment by ajwfrost** [comment]
@marchbold thanks for checking on this. The only issue I see though is that the manifestadditions has other `<activity>` elements which have intent filters (see the below) - presumably defined within an ANE, I suspect that we'd block the use of the `exported` attribute on those too.

```
					<activity
						android:name="com.tencent.midas.qq.APMidasQQWalletActivity"
						android:launchMode="singleTop"
						android:theme="@android:style/Theme.Translucent.NoTitleBar"
						android:configChanges="orientation|keyboardHidden"
						android:exported="false|true" >
						<intent-filter >
							<action android:name="android.intent.action.VIEW"/>
							<category android:name="android.intent.category.BROWSABLE"></category>
							<category android:name="android.intent.category.DEFAULT"></category>
							<data android:scheme="qwallet100703379"></data>
						</intent-filter>
					</activity>
```

So yes, ideally we would ensure the main activity remains as-is but allow other activities to be defined and customized... (and maybe even to check that the values are correct boolean strings per your other note above..)

thanks

**Comment by marchbold** [comment]
Fairly sure that all works already , we have been adding activities with intent filters for years...

**Comment by kubiyuan** [comment]
Packaging issues have been resolved

Adding 'preventDeviceModelAccess' tag in the 'android' section of the application descriptor to stop Build.MODEL call


Is it correct to write?

<img width="532" alt="企业微信截图_1d6f0827-d6db-4022-923b-6ed1c4297ee2" src="https://user-images.githubusercontent.com/54053483/182311923-72ab5656-76f4-43d8-9a77-de4e51c8ce6c.png">
